department-of-veterans-affairs/caseflow
# JHoang/APPEALS-32914-2

Resolves [32914](https://jira.devops.va.gov/browse/APPEALS-32914?workflowName=APPEAL+Story+Workflow&stepId=20)

# Description

- Refactored seed file for Correspondence so that it doesn't generate random data.
- Added more Correspondence seed data -> specifically one of each type from the `package_document_types`, `correspondence_types`, and `cmp_queue` tables.

## Acceptance Criteria

## Testing Plan
1. Go to [32914/Test Plan Link](https://jira.devops.va.gov/browse/APPEALS-32914?workflowName=APPEAL+Story+Workflow&stepId=20)

or

1. Open rails console `rails c`
2. `RequestStore[:current_user] = User.system_user`
3. `Dir[Rails.root.join("db/seeds/*.rb")].sort.each { |f| require f }`
4. `Seeds::Correspondence.new.seed!`
5. Observe that 128 Correspondences should be created in total

![correspondences 1](https://github.com/department-of-veterans-affairs/caseflow/assets/111081469/8e8552b7-332a-4d8f-bdaf-a5552eb97726)
![correspondences_2](https://github.com/department-of-veterans-affairs/caseflow/assets/111081469/bef269a8-54a1-46e5-aef2-7f87278ecef8)
![correspondence_documents](https://github.com/department-of-veterans-affairs/caseflow/assets/111081469/f5b6fc33-756f-41a9-9d99-32830d5c0dbc)
<img width="581" alt="correspondence_count" src="https://github.com/department-of-veterans-affairs/caseflow/assets/111081469/b5be5437-767b-464e-ad02-da6efbf297bb">


# Best practices
## Code Documentation Updates

## Tests
### Test Coverage
Did you include any test coverage for your code? Check below:

### Code Climate
Your code does not add any new code climate offenses? If so why?

## Monitoring, Logging, Auditing, Error, and Exception Handling Checklist

### Monitoring

### Logging

### Auditing

### Error Handling

### Exception Handling
